### PR TITLE
add 2.0.0-beta.21 changelog release entry

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Adds 2.0.0-beta.2 1CHANGELOG release entry

### Changelog

```
<changelog></changelog>
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,31 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+## Mapbox Navigation SDK 2.0.0-beta.21 - July 28, 2021
+
+For details on how v2 differs from v1 and guidance on migrating from v1 of the Mapbox Navigation SDK for Android to the v2 public preview, see [2.0 Navigation SDK Migration Guide](https://github.com/mapbox/mapbox-navigation-android/wiki/2.0-Navigation-SDK-Migration-Guide).
+
+### Changelog
+#### Bug fixes and improvements
+- Add `BuildingHighlightObserver` for exposing map data used to highlight a building. Add example showing extrude all building along with a highlighted one. [#4420](https://github.com/mapbox/mapbox-navigation-android/pull/4420)
```
Comment (Guardiola31337) [inline CHANGELOG.md:11]:
```suggestion
- Added `BuildingHighlightObserver` for exposing map data used to highlight a building. Added example showing extrude all building along with a highlighted one. [#4420](https://github.com/mapbox/mapbox-navigation-android/pull/4420)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,31 @@
...
+- :warning: Removed `accessToken` from `RouteOptions`, the token provided via `NaivgationOptions` is now always used for route requests. This also simplifies serialization and deserialization of route objects. Refs [`mapbox-java` `v6.0.0-alpha2`](https://github.com/mapbox/mapbox-java/releases/tag/v6.0.0-alpha.2). [#4670](https://github.com/mapbox/mapbox-navigation-android/pull/4670)
+- Add geometry to `Maneuver` object. [#4653](https://github.com/mapbox/mapbox-navigation-android/pull/4653)
```
Comment (Guardiola31337) [inline CHANGELOG.md:13]:
```suggestion
- Added geometry to `Maneuver` object. [#4653](https://github.com/mapbox/mapbox-navigation-android/pull/4653)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,31 @@
...
+  - Exposed `Factory` to allow developers to create `Maneuver` object.
+  - Added `Point` as another argument to `Maneuver`
+- Removed access token from `MapboxHistoryReader`. [#4667](https://github.com/mapbox/mapbox-navigation-android/pull/4667)
+  - Add `ReplayHistorySetRouteMapper` to map set route events from the `ReplayHistoryMapper`.
+  - This change has been reverted, only the token removal is a valid change.
```
Comment (Guardiola31337) [inline CHANGELOG.md:18]: This is confusing 🤔 

cc @kmadsen

**Review by Guardiola31337 — COMMENTED** [review]
Left a couple of comments @abhishek1508

File: CHANGELOG.md (hunk 3 above)
Comment (kmadsen) [reply CHANGELOG.md:18]: We can remove these changes from the release. 


```
  - Added `ReplayHistorySetRouteMapper` to map set route events from the `ReplayHistoryMapper`.
  - This change has been reverted, only the token removal is a valid change.
  ```

And then change this to include a couple pull requests.
```
- Removed access token from `MapboxHistoryReader`. [#4667](https://github.com/mapbox/mapbox-navigation-android/pull/4667), [#4670](https://github.com/mapbox/mapbox-navigation-android/pull/4670) 
```

### Explanation

We iterated on the solution within this release. Mentioning classes that are not in this release, or a previous release - creates confusion. The difference between 2.0.0-beta-20 and 2.0.0-beta-12 is: We removed the access token from `ReplayHistoryMapper`. And removed the access token from `RouteOptions`

Comment (kmadsen) [reply CHANGELOG.md:18]:
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,31 @@
...
+
+### Changelog
+#### Bug fixes and improvements
+- Added `BuildingHighlightObserver` for exposing map data used to highlight a building. Add example showing extrude all building along with a highlighted one. [#4420](https://github.com/mapbox/mapbox-navigation-android/pull/4420)
+- :warning: Removed `accessToken` from `RouteOptions`, the token provided via `NaivgationOptions` is now always used for route requests. This also simplifies serialization and deserialization of route objects. Refs [`mapbox-java` `v6.0.0-alpha2`](https://github.com/mapbox/mapbox-java/releases/tag/v6.0.0-alpha.2). [#4670](https://github.com/mapbox/mapbox-navigation-android/pull/4670)
+- Added geometry to `Maneuver` object. [#4653](https://github.com/mapbox/mapbox-navigation-android/pull/4653)
+  - Exposed `Factory` to allow developers to create `Maneuver` object.
+  - Added `Point` as another argument to `Maneuver`
+- Removed access token from `MapboxHistoryReader`. [#4667](https://github.com/mapbox/mapbox-navigation-android/pull/4667)
+  - Added `ReplayHistorySetRouteMapper` to map set route events from the `ReplayHistoryMapper`.
```
Comment (kmadsen) [inline CHANGELOG.md:17]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,31 @@
...
+
+### Changelog
+#### Bug fixes and improvements
+- Added `BuildingHighlightObserver` for exposing map data used to highlight a building. Add example showing extrude all building along with a highlighted one. [#4420](https://github.com/mapbox/mapbox-navigation-android/pull/4420)
+- :warning: Removed `accessToken` from `RouteOptions`, the token provided via `NaivgationOptions` is now always used for route requests. This also simplifies serialization and deserialization of route objects. Refs [`mapbox-java` `v6.0.0-alpha2`](https://github.com/mapbox/mapbox-java/releases/tag/v6.0.0-alpha.2). [#4670](https://github.com/mapbox/mapbox-navigation-android/pull/4670)
+- Added geometry to `Maneuver` object. [#4653](https://github.com/mapbox/mapbox-navigation-android/pull/4653)
+  - Exposed `Factory` to allow developers to create `Maneuver` object.
+  - Added `Point` as another argument to `Maneuver`
+- Removed access token from `MapboxHistoryReader`. [#4667](https://github.com/mapbox/mapbox-navigation-android/pull/4667)
```
Comment (kmadsen) [inline CHANGELOG.md:16]:
```suggestion
- Removed access token from `MapboxHistoryReader`. [#4667](https://github.com/mapbox/mapbox-navigation-android/pull/4667), [#4670](https://github.com/mapbox/mapbox-navigation-android/pull/4670) 

```

**Review by Guardiola31337 — APPROVED** [review]

**Review by kmadsen — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,30 @@
...
+### Changelog
+#### Bug fixes and improvements
+- Added `BuildingHighlightObserver` for exposing map data used to highlight a building. Add example showing extrude all building along with a highlighted one. [#4420](https://github.com/mapbox/mapbox-navigation-android/pull/4420)
+- :warning: Removed `accessToken` from `RouteOptions`, the token provided via `NaivgationOptions` is now always used for route requests. This also simplifies serialization and deserialization of route objects. Refs [`mapbox-java` `v6.0.0-alpha2`](https://github.com/mapbox/mapbox-java/releases/tag/v6.0.0-alpha.2). [#4670](https://github.com/mapbox/mapbox-navigation-android/pull/4670)
+- Added geometry to `Maneuver` object. [#4653](https://github.com/mapbox/mapbox-navigation-android/pull/4653)
+  - Exposed `Factory` to allow developers to create `Maneuver` object.
+  - Added `Point` as another argument to `Maneuver`
+- Removed access token from `MapboxHistoryReader`. [#4667](https://github.com/mapbox/mapbox-navigation-android/pull/4667), [#4670](https://github.com/mapbox/mapbox-navigation-android/pull/4670)
+
+
```
Comment (kmadsen) [inline CHANGELOG.md:18]: extra line nitpick
```suggestion
```

**Pull request merged**